features/graph/rest/src/main/java/org/opennms/netmgt/graph/rest/api/GraphSearchRestService.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.graph.rest.api;
+
+import java.util.List;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.opennms.netmgt.graph.api.generic.GenericVertex;
+import org.opennms.netmgt.graph.api.search.SearchSuggestion;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Path("graphs/search")
```

Comment:
this doesn't seem to work, neither does http://localhost:8980/opennms/api/v2/graphs
Not sure if it's just the wrong URL or something else. @mvrueden would you know that by heart before I start investigating?

Reply:
Not 100% sure, but its either a dependency issue or a clash from GraphRestService. 
I will take a closer look.

Reply:
:+1: thanks

Reply:
There were several issues: 
 - `@RequestParam` is spring web stuff, whereas we need `@QueryParam` which is a `javax.ws.rs.` class.
 - `ApplicationTopologyProvider` did not wire in the `GraphSearchService` correctly
 - Some other places also did not wire in stuff correctly.
 - The feature `opennms-graphs` must be installed manually, as well as the according providers.

I fixed it, so after installing the correct features rest etc works. But only tested get suggestions.